### ISSUE TITLE
1.10.4: "'ShowBase' is not defined" on DirectButton.destroy()

### ISSUE THREAD

**Issue by ArsThaumaturgis** [panda3d/panda3d#708]
I seem to have encountered a bug in 1.10.4. The code that prompts this worked under a development build (which had a 1.11 version number, I think), and seems innocuous enough--it just involves destroying a DirectButton.

A quick look at the history for "DirectGuiBase.py" indicates that the problem comes from the latest commit to affect that file. That commit removed the importation of ShowBase, in favour of "ShowBaseGlobal"--but the line that causes the crash (1033) still attempts to access ShowBase.

Simply changing "ShowBase" to "ShowBaseGlobal" on line 1033 of "DirectGuiBase.py" seems to fix the problem!

The full error:

```text
Traceback (most recent call last):
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/EventManager.py", line 49, in eventLoopTask
    self.doEvents()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/EventManager.py", line 43, in doEvents
    processFunc(dequeueFunc())
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/EventManager.py", line 99, in processEvent
    messenger.send(eventName, paramList)
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/Messenger.py", line 333, in send
    self.__dispatch(acceptorDict, event, sentArgs, foundWatch)
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/Messenger.py", line 418, in __dispatch
    result = method (*(extraArgs + sentArgs))
  File "/home/thaumaturge/Documents/My Game Projects/BugHunt/bugHunt.py", line 18, in destroyBtn
    self.mew.destroy()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/gui/DirectFrame.py", line 62, in destroy
    DirectGuiWidget.destroy(self)
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/gui/DirectGuiBase.py", line 1033, in destroy
    if hasattr(ShowBase, 'guiItems'):
NameError: name 'ShowBase' is not defined
:task(error): Exception occurred in PythonTask eventManager
Traceback (most recent call last):
  File "/home/thaumaturge/Documents/My Game Projects/BugHunt/bugHunt.py", line 22, in <module>
    app.run()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/ShowBase.py", line 3124, in run
    self.taskMgr.run()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/task/Task.py", line 531, in run
    self.step()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/task/Task.py", line 485, in step
    self.mgr.poll()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/EventManager.py", line 49, in eventLoopTask
    self.doEvents()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/EventManager.py", line 43, in doEvents
    processFunc(dequeueFunc())
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/EventManager.py", line 99, in processEvent
    messenger.send(eventName, paramList)
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/Messenger.py", line 333, in send
    self.__dispatch(acceptorDict, event, sentArgs, foundWatch)
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/showbase/Messenger.py", line 418, in __dispatch
    result = method (*(extraArgs + sentArgs))
  File "/home/thaumaturge/Documents/My Game Projects/BugHunt/bugHunt.py", line 18, in destroyBtn
    self.mew.destroy()
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/gui/DirectFrame.py", line 62, in destroy
    DirectGuiWidget.destroy(self)
  File "/home/thaumaturge/.local/lib/python3.6/site-packages/direct/gui/DirectGuiBase.py", line 1033, in destroy
    if hasattr(ShowBase, 'guiItems'):
NameError: name 'ShowBase' is not defined

Process finished with exit code 1
```

A simple test-program follows that demonstrates the issue. Just run the program, which constructs a DirectButton, then press the space-bar to attempt to delete it. On my machine the program crashes and prints the above error--unless I change "ShowBase" to "ShowBaseGlobal" in "DirectGuiBase.py", in which case it works as expected.

```python
from direct.showbase import ShowBase as showBase
from direct.gui.DirectGui import DirectButton

class Game(showBase.ShowBase):

    def __init__(self):
        showBase.ShowBase.__init__(self)

        self.mew = DirectButton(text = "Kittens",
                           scale = 0.07)

        self.accept("escape", base.userExit)
        self.accept("space", self.destroyBtn)

    def destroyBtn(self):
        self.mew.destroy()
        self.mew = None

app = Game()
app.run()
```

**Comment by rdb** [comment]
Thank you for reporting this (embarrassing) regression.  This is already fixed, and I'm in the process of releasing 1.10.4.1 with the fix included.

**Comment by ArsThaumaturgis** [comment]
Hahah, fair enough! That was nicely quick! ^_^
